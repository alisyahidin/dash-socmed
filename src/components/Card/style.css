.my-card {
  border: none !important;
  box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.1);
}

.my-card__body {
  display: flex;
  flex-direction: column;
}

.my-card__body--left {
  align-items: flex-start;
}

.my-card__body--center {
  align-items: center;
}

.my-card__body--right {
  align-items: flex-end;
}