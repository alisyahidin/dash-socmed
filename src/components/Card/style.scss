.my-card {
  border: none !important;
  box-shadow: 0px 1px 1px rgba(0, 0, 0, 0.1);
  &__body {
    display: flex;
    flex-direction: column;
    &--left {
      align-items: flex-start;
    }
    &--center {
      align-items: center;
    }
    &--right {
      align-items: flex-end;
    }
  }
}