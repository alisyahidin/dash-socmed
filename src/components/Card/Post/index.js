import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { Link } from 'react-router-dom'
import { Row, Col } from 'react-bootstrap'

import { openModal } from '../../../actions/singlePost'
import './style.scss'

const Post = ({author, id, title, body, openModal, ...props}) => {
  return (
    <>
      <Row className="post mx-sm-1">
        <Col className="p-sm-0 d-flex flex-column justify-content-center align-items-start">
          <h5 onClick={() => openModal(true, id)} className="post__title mb-sm-2">{title}</h5>
          <span>
            by&nbsp;-&nbsp;
            <Link to={`/users/${author.id}`} className="author d-inline-block mb-sm-1">
              <b className="mr-sm-1">
                {author.name}
              </b>
              <span className="author__username mr-sm-2">
                @{author.username}
              </span>
            </Link>
          </span>
          <div onClick={() => openModal(true, id)} className="post__body">
            <p className="mb-sm-2">{body}</p>
          </div>
        </Col>
      </Row>
    </>
  )
}

Post.propTypes = {
  id: PropTypes.number.isRequired,
  author: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
}

Post.defaultProps = {
  id: 0,
  author: {
    id: 1,
    name: 'Guest',
    username: 'guest'
  },
  title: 'Title',
  body: 'Post',
}

const mapStateToProps = state => ({ ...state })

const mapDispatchToProps = dispatch =>
  bindActionCreators({
    openModal
  }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(Post)
