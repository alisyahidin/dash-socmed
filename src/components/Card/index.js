import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { Card as BsCard } from 'react-bootstrap'
import './style.css'

const Card = ({children, className, align}) => (
  <BsCard className={classNames("my-card", className)}>
    <BsCard.Body
      className={`my-card__body my-card__body--${align}`}
    >
      {children}
    </BsCard.Body>
  </BsCard>
)

Card.propTypes = {
  children: PropTypes.any.isRequired,
  className: PropTypes.string,
  align: PropTypes.oneOf(['left', 'center', 'right']),
}

Card.defaultProps = {
  align: 'center',
  children: 'Card'
}

export default Card