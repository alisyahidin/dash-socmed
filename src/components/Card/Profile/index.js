import React from 'react'
import PropTypes from 'prop-types'
import Avatar from 'react-avatar'
import Card from '../'
import './style.css'

const Profile = ({name, email}) => (
  <Card align="center">
    <Avatar round size="70px" name={name} className="profile-avatar" />
    <h4>{name}</h4>
    <p className="profile-email">{email}</p>
  </Card>
)

Profile.propTypes = {
  name: PropTypes.string.isRequired,
  email: PropTypes.string.isRequired,
}

Profile.defaultProps = {
  name: 'Guest',
  email: 'quest@email.com',
}

export default Profile