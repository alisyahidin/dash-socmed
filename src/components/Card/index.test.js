import React from 'react'
import { shallow } from 'enzyme'
import Card from './index'

it('render <Card /> correctly', () => {
  const component = shallow(<Card>Test</Card>)

  expect(component).toMatchSnapshot()
})