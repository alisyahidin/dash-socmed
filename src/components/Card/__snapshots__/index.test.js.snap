// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <Card /> correctly 1`] = `
<ForwardRef(Bootstrap(Card))
  className="my-card"
>
  <ForwardRef(Bootstrap(CardBody))
    className="my-card__body my-card__body--center"
  >
    Test
  </ForwardRef(Bootstrap(CardBody))>
</ForwardRef(Bootstrap(Card))>
`;
